nautobot/docs/development/apps/migration/migrating-an-app-from-v1.md

```diff
@@ -0,0 +1,191 @@
+
+#### Convert Relationship Type between Prefix and VRF to Many to Many
+
+[Prefixes](../../../user-guide/core-data-model/ipam/prefix.md) now no longer has a ForeignKey to [VRF](../../../user-guide/core-data-model/ipam/vrf.md). Instead, the Many to Many relationship is now defined on the VRF side as `VRF.prefixes`. VRF is no longer assigned to an IPAddress and is now on the parent Prefix. It is now a M2M relationship between the VRF and Prefix. VRF is also no longer a uniqueness constraint on the Prefix. Namespace is used instead. There is a default `Global` Namespace that all Prefixes are migrated into from 1.x.
+
+## Code Updates
+
+### Update Code Import Locations
+
+Most changes in code location arise from the merging of the `nautobot.utilities` module into the `nautobot.core` module.
+
+??? info "Full table of code location changes"
+    {data-table user-guide/administration/upgrading/from-v1/tables/v2-code-location-changes.yaml}
+
+### Replace PluginMenuItem with NavMenuItem
+
+In your app's `navigation.py` file. If you are still using `PluginMenuItem` from `nautobot.extras.plugin`, you should replace those code with `NavMenuGroup`, `NavMenuItem`, and `NavMenuTab` from `nautobot.apps.ui`.
+
+For example:
+
+Before:
+
+```python
+
+    from nautobot.extras.plugins import PluginMenuItem
+
+    menu_items = (
+        PluginMenuItem(
+            link="plugins:your_app:dashboard",
+            link_text="Dashboard",
+            permissions=["your_app.view_sync"],
+        ),
+        PluginMenuItem(
+            link="plugins:your_app:sync_list",
+            link_text="History",
+            permissions=["your_app.view_sync"],
+        ),
+        PluginMenuItem(
+            link="plugins:your_app:synclogentry_list",
+            link_text="Logs",
+            permissions=["your_app.view_synclogentry"],
+        ),
+    )
+
+```
+
+After:
+
+```python
+
+from nautobot.apps.ui import NavMenuGroup, NavMenuItem, NavMenuTab
+
+
+items = [
+    NavMenuItem(
+        link="plugins:your_app:dashboard",
+        name="Dashboard",
+        permissions=["your_app.view_sync"],
+    ),
+    NavMenuItem(
+        link="plugins:your_app:sync_list",
+        name="History",
+        permissions=["your_app.view_sync"],
+    ),
+    NavMenuItem(
+        link="plugins:your_app:synclogentry_list",
+        name="Logs",
+        permissions=["your_app.view_synclogentry"],
+    ),
+]
+
+menu_items = (
+    NavMenuTab(
+        name="Plugins",
+        groups=(NavMenuGroup(name="Your App", weight=1000, items=tuple(items)),),
+    ),
+)
+
+```
+
+### Replace DjangoFilterBackend with NautobotFilterBackend
+
+If your Custom `FilterBackend` class is derived from `DjangoFilterBackend`, you should replace `DjangoFilterBackend` with `NautobotFilterBackend`.
```

Comment:
why? maybe a bit of context of what this replacement is recommended would be good